src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const state2 = Object.assign({}, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state2, action.extraData);
+        result.push({ ...state2 });
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          delete state2[key];
+        }
+        result.push({ ...state2 });
+        break;
+
+      case 'clear':
+        for (const key in state2) {
+          delete state2[key];
+        }
+        result.push({ ...state2 });
+        break;
+    }
+  }
```

Comment:
`result.push({ ...state2 });` repeats 3 times - do it after switch
```suggestion
    switch (type) {
      case 'addProperties':
        Object.assign(state2, action.extraData);
        break;

      case 'removeProperties':
        const { keysToRemove } = action;

        for (const key of keysToRemove) {
          delete state2[key];
        }
        break;

      case 'clear':
        for (const key in state2) {
          delete state2[key];
        }
        break;
    }
    result.push({ ...state2 });
  }
```